x-pack/plugins/ml/public/application/license/expired_warning.tsx

```diff
@@ -20,8 +20,15 @@ export function showExpiredLicenseWarning() {
     });
     // Only show the banner once with no way to dismiss it
     const overlays = getOverlays();
+    const theme = getTheme();
```

Comment:
We actually can avoid using dependency cache here by passing overlays and a theme as arguments. First to `MlClientLicense` constructor and then to this function.

Reply:
`MlClientLicense` would need to keep a local reference to `overlays` and `theme` for use later, which is no different to the dependency cache keeping references to them.
At least with the dependency cache they are all in the same place.

Reply:
Fair point. @jgowdyelastic and I discussed this and decided that keeping a copy in the cache rather than a copy in the licence is simpler for now.